### PR TITLE
Clean up processor documentation

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Also expose ctors for the EventArgs types for mocking.

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs [hunk 1]
```diff
@@ -31,23 +31,18 @@ public class ProcessMessageEventArgs : EventArgs
         /// Initializes a new instance of the <see cref="ProcessMessageEventArgs"/> class.
         /// </summary>
         ///
-        /// <param name="message"></param>
-        /// <param name="receiver"></param>
-        /// <param name="cancellationToken"></param>
-        internal ProcessMessageEventArgs(ServiceBusReceivedMessage message, ServiceBusReceiver receiver, CancellationToken cancellationToken)
+        /// <param name="message">The message to be processed.</param>
+        /// <param name="receiver">The receiver instance that can be used to perform message settlement.</param>
+        /// <param name = "cancellationToken" > The processor's <see cref="System.Threading.CancellationToken"/> instance which will be cancelled in the event that <see cref="ServiceBusProcessor.StopProcessingAsync"/> is called.
```
Comment (MiYanni) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs:36]: I'm guessing the extra spaces in the param tag don't affect it, right?

**Review by MiYanni — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs:36]: I don't think so, but not sure why that happened.

**Pull request merged**